runtime-modules/working-team/src/lib.rs

```diff
@@ -774,19 +777,43 @@ decl_module! {
             //
 
             let status_text_hash = status_text
-            .map(|status_text| {
-                    let hashed = T::Hashing::hash(&status_text);
+                .map(|status_text| {
+                        let hashed = T::Hashing::hash(&status_text);
 
-                    hashed.as_ref().to_vec()
-                })
-            .unwrap_or_default();
+                        hashed.as_ref().to_vec()
+                    })
+                .unwrap_or_default();
 
             // Update the status text hash.
             <StatusTextHash<I>>::put(status_text_hash.clone());
 
             // Trigger event
             Self::deposit_event(RawEvent::StatusTextChanged(status_text_hash));
         }
+
+        /// Transfers specified amount to any account.
+        /// Requires leader origin.
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn spend_from_budget(
+            origin,
+            account_id: T::AccountId,
+            amount: BalanceOfCurrency<T>,
+            rationale: Option<Vec<u8>>,
+        ) {
+            // Ensure team leader privilege.
+            checks::ensure_origin_is_active_leader::<T,I>(origin)?;
+
+            ensure!(amount > Zero::zero(), Error::<T, I>::CannotSpendZero);
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            Self::try_to_pay_from_budget(&account_id, amount)?;
```

Comment:
I don't love this ? check after mutation check. Can we not avoid this? if the check creates imbalance for example?